### PR TITLE
mrtrix3: new package

### PR DESCRIPTION
**Pull request by akthoma**
adding new package mrtrix3

### CODE REVIEW

File: var/spack/repos/builtin/packages/mrtrix3/package.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    depends_on('python@2.7:', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('mesa-glu')
+    depends_on('qt+opengl@4.7:')
+    depends_on('eigen')
+    depends_on('zlib')
+    depends_on('libtiff')
+    depends_on('fftw')
+
+    conflicts('%gcc@7:')
```
Comment (junghans) [inline var/spack/repos/builtin/packages/mrtrix3/package.py]: What is the problem with gcc-7?

Comment (akthoma) [reply var/spack/repos/builtin/packages/mrtrix3/package.py]: Doesn't compile. Lots of errors mentioning deprecation--I've experienced a lot of problems with different software having these kind of errors with gcc 7 but when compiling with 6.x or below they disappear. I can paste the output of the spack installer errors if you would like to take a closer look.

Comment (junghans) [reply var/spack/repos/builtin/packages/mrtrix3/package.py]: Ok, please open an issue at https://github.com/MRtrix3/mrtrix3 and post the details of the compile issue there, then refer to that issue in this spackage and weaken the conflict to `conflicts('%gcc@7:', when='@2017-09-25')`. Long term package should support gcc-7 and the only way to make that happen is to let upstream know.

**Comment by akthoma** [comment]
https://github.com/MRtrix3/mrtrix3/pull/1041

The errors I was getting are already known about and referenced in the pull request above on the mrtrix3 github.

> I agree that both the 'previous definition' and 'dereferencing type-punned pointer will break strict-aliasing rules' warnings are probably compiler bugs that'll hopefully go away at some point. They don' happen with clang++ anyway.

**Pull request merged**